plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -130,7 +150,14 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 	// Query is not run if Database version does not match the query version.
 	for i := range p.Query {
 		sql_query = p.Query[i].Sqlquery
+		if sql_query == "" {
+			sql_query, err = ReadQueryFromFile(p.Query[i].Script)
```

Comment:
I agree with Daniel, move this into `Init()` and save it on the struct so the file isn't read each call to `Gather`

```go
func (p *Postgresql) Init() error {
    // load the file and save onto the Postgresql struct.
    // return an error if the file is unreadable or configuration is invalid
}
```